Assets/MRTK/SDK/Experimental/SpatialAwareness/SurfaceMeshesToPlanes.cs

```diff
@@ -0,0 +1,473 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.SpatialAwareness;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using UnityEngine;
+
+#if PLANE_FINDING_PRESENT
+using Microsoft.MixedReality.Toolkit.SpatialAwareness.Processing;
+#endif
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.SpatialAwareness
+{
+    struct PlaneWithType
+    {
+        public GameObject Plane;
+        public SpatialAwarenessSurfaceTypes Type;
+    }
+
+    /// <summary>
+    /// SurfaceMeshesToPlanes will find and create planes based on the meshes returned by the SpatialMappingManager's Observer.
+    /// </summary>
+    public class SurfaceMeshesToPlanes : MonoBehaviour
+    {
+#region Public properties
+        [SerializeField]
+        [Tooltip("Empty game object used to contain all planes created by the SurfaceToPlanes class")]
+        private GameObject planesParent;
+
+        /// <summary>
+        /// Empty game object used to contain all planes created by the SurfaceToPlanes class
+        /// </summary>
+        public GameObject PlanesParent
+        {
+            get => planesParent;
+            set => planesParent = value;
+        }
+
+        [SerializeField]
+        [Tooltip("The physics layer for planes to be set to")]
+        private int physicsLayer;
+
+        /// <summary>
+        /// The physics layer for planes to be set to
+        /// </summary>
+        public int PhysicsLayer
```

Comment:
This all feels a little observer-y 😬 do we think this is the right path, or should we go ahead and add a data provider observer instead?